### PR TITLE
Adds BlameReport and corresponding experiment.

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/reports/blame_report.py  3
- tests/data/test_blame_report.py  3
- varats/experiments/blame_report_experiment.py  3
- varats/experiments/blame_experiment.py  3
         

Clones added
============
- varats/experiments/git_blame_annotation_report.py  4
- varats/experiments/blame_report_experiment.py  3
- varats/experiments/blame_experiment.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4445398&bid=15031319)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/experiments/blame_experiment.py  3
- varats/experiments/blame_report_experiment.py  3
- tests/data/test_blame_report.py  3
- varats/data/reports/blame_report.py  3
         

Clones added
============
- varats/experiments/blame_experiment.py  2
- varats/experiments/blame_report_experiment.py  3
- varats/experiments/git_blame_annotation_report.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4445398&bid=15031319)

**Comment by vulder** [comment]
the clones will be cleaned up with a separate commit

File: tests/data/test_blame_report.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                         '48f8ed5347aeb9d54e7ea041b1f8d67ffe74db33')
+        self.assertEqual(c_interaction_list[1].amount, 5)
+
+        cxx_interaction_list = self.func_entry_cxx.interactions
+        self.assertEqual(cxx_interaction_list, [])
+
+
+class TestBlameReport(unittest.TestCase):
+    """
+    Test if a blame report is correctly reconstruction from yaml.
```
Comment (boehmseb) [inline tests/data/test_blame_report.py]: typo: reconstructed

File: varats/data/reports/blame_report.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+from pathlib import Path
+import yaml
+
+from varats.data.report import BaseReport, FileStatusExtension
+from varats.data.version_header import VersionHeader
+
+
+class BlameInstInteractions():
+    """
+    An interaction of between other commits an a base.
```
Comment (boehmseb) [inline varats/data/reports/blame_report.py]: ?

File: varats/experiments/blame_report_experiment.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
+"""
+Implements the blame report experiment.
+
+The experiment generates analyses a project and generates a BlameReports with
+VaRA.
```
Comment (boehmseb) [inline varats/experiments/blame_report_experiment.py]: That sentence is broken.

**Review by boehmseb — CHANGES_REQUESTED** [review]
A couple of docstrings are broken and therefore hard to understand. Otherwise, this looks fine.

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- varats/experiments/blame_experiment.py  3
- tests/data/test_blame_report.py  3
- varats/data/reports/blame_report.py  3
- varats/experiments/blame_report_experiment.py  3
         

Clones added
============
- varats/experiments/blame_experiment.py  2
- varats/experiments/git_blame_annotation_report.py  4
- varats/experiments/blame_report_experiment.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4445398&bid=15031319)

**Review by boehmseb — APPROVED** [review]

**Pull request merged**